packages/react-components/src/components/Search/Search.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+import cx from 'clsx';
+import {
+  Search as SearchIcon,
+  Close,
+} from '@livechat/design-system-icons/react/material';
+import { Icon, IconTypeName } from '../Icon';
+import { Loader } from '../Loader';
+
+import styles from './Search.module.scss';
+import { KeyCodes } from '../../utils/keyCodes';
+
+const baseClass = 'search';
+const inputBaseClass = `${baseClass}__input`;
+
+export const enum SearchSize {
+  Compact = 'compact',
+  Medium = 'medium',
+  Large = 'large',
+}
+
+export interface ISearchProps {
+  isControlledSubmit?: boolean;
+  isCollapsable?: boolean;
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  placeholder?: string;
+  size?: SearchSize;
+  value?: string;
+  onChange: (value: string) => void;
+}
+
+export const Search: React.FC<ISearchProps> = ({
+  isControlledSubmit,
+  isCollapsable,
+  isDisabled,
+  isLoading,
+  placeholder = 'Search ...',
+  size = SearchSize.Compact,
+  value,
+  onChange,
+}) => {
+  const [searchValue, setSearchValue] = React.useState<string>(value || '');
+  const [isCollapsed, setIsCollapsed] = React.useState<boolean>(true);
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const mergedClassNames = cx(
+    styles[`${inputBaseClass}`],
+    styles[`${inputBaseClass}--${size}`],
+    (isDisabled || isLoading) && styles[`${inputBaseClass}--disabled`],
+    isCollapsable && styles[`${inputBaseClass}--collapsable`],
+    !isCollapsed && styles[`${inputBaseClass}--collapsable--open`]
+  );
+
+  const handleOnChange = (e: React.FormEvent<HTMLInputElement>) => {
+    const value = e.currentTarget.value;
+
+    setSearchValue(value);
+
+    if (isControlledSubmit) {
+      return;
+    }
+
+    return onChange(value);
```

Comment:
I don't buy it :) Could you elaborate more on this "feature"?

Reply:
`isControlledSubmit` is a flag to control input behavior, disabling firing onChange automatically.